huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go

```diff
@@ -0,0 +1,512 @@
+	}
+
+	checkCreateErr := checkFlinkJobRunResult(ctx, client, rst.Job.JobId, d.Timeout(schema.TimeoutCreate))
+	if checkCreateErr != nil {
+		return diag.FromErr(checkCreateErr)
+	}
+	return resourceFlinkJarJobRead(ctx, d, meta)
+}
+
+func resourceFlinkJarJobRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.DliV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DLI v1 client, err=%s", err)
+	}
+	id, aErr := strconv.Atoi(d.Id())
+	if aErr != nil {
+		return fmtp.DiagErrorf("the DLI flink job_id must be number. actual id=%s", d.Id())
+	}
+
+	detailRsp, dErr := flinkjob.Get(client, id)
+	if dErr != nil {
+		return fmtp.DiagErrorf("Error query DLI flink jar job %q:%s", id, dErr)
+	}
+
+	if detailRsp != nil && !detailRsp.IsSuccess {
+		return fmtp.DiagErrorf("Error query DLI flink jar job: %s", detailRsp.Message)
+	}
+	detail := detailRsp.JobDetail
+	mErr := multierror.Append(
+		d.Set("name", detail.Name),
+		d.Set("description", detail.Desc),
+		d.Set("queue_name", detail.QueueName),
+		d.Set("main_class", detail.MainClass),
+		d.Set("entrypoint", detail.JobConfig.Entrypoint),
+		d.Set("entrypoint_args", detail.EntrypointArgs),
+		d.Set("dependency_jars", detail.JobConfig.DependencyJars),
+		d.Set("dependency_files", detail.JobConfig.DependencyFiles),
+		d.Set("feature", detail.JobConfig.Feature),
+		d.Set("flink_version", detail.JobConfig.FlinkVersion),
+		d.Set("image", detail.JobConfig.Image),
+		d.Set("cu_number", detail.JobConfig.CuNumber),
+		d.Set("parallel_number", detail.JobConfig.ParallelNumber),
+		d.Set("obs_bucket", detail.JobConfig.ObsBucket),
+		d.Set("log_enabled", detail.JobConfig.LogEnabled),
+		d.Set("smn_topic", detail.JobConfig.SmnTopic),
+		d.Set("restart_when_exception", detail.JobConfig.RestartWhenException),
+		d.Set("manager_cu_number", detail.JobConfig.ManagerCuNumber),
+		d.Set("tm_cus", detail.JobConfig.TmCus),
+		d.Set("tm_slot_num", detail.JobConfig.TmSlotNum),
+		d.Set("resume_checkpoint", detail.JobConfig.ResumeCheckpoint),
+		d.Set("resume_max_num", detail.JobConfig.ResumeMaxNum),
+		d.Set("checkpoint_path", detail.JobConfig.CheckpointPath),
+		d.Set("runtime_config", utils.TagsToMap(parseConfig(detail.JobConfig.RuntimeConfig))),
+		d.Set("status", detail.Status),
+	)
+	if setSdErr := mErr.ErrorOrNil(); setSdErr != nil {
+		return fmtp.DiagErrorf("Error setting vault fields: %s", setSdErr)
+	}
+
+	return nil
+}
+
+func resourceFlinkJarJobDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.DliV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DLI v1 client, err=%s", err)
+	}
+
+	jobId, aErr := strconv.Atoi(d.Id())
+	if aErr != nil {
+		return fmtp.DiagErrorf("the DLI flink job_id must be number. actual id=%s", d.Id())
+	}
```

Comment:
ditto. this step is unnecessary.

Reply:
keep it.  import will use this function.